pagopa/pn-frontend
# fix/PN-9145: fix on catching error for duplicated protocol number from FE

## Short description
Fix on createNewNotification in case of error for duplicated protocol number when creating a notification.

## List of changes proposed in this pull request
- fixed createNotification method to catch error for duplicated protocol number in this part too (the error was already managed showing the toast of error)
- implemented unit test "notification failed for duplicated protocol number"

## How to test
Login in as PA, click on "invia una notifica", create a notification with a protocol number. Try to create a new notification with the same protocol number of the previous one. Check a toast of error appear.
